worker/include/Utils.hpp

```diff
@@ -349,6 +349,46 @@ namespace Utils
 				return false;
 		}
 	};
+
+	template<typename T>
+	class ObjectPool
+	{
+	public:
+		~ObjectPool()
+		{
+			for (auto ptr : this->pool)
+			{
+				std::free(ptr);
+			}
+		}
+
+		// Get pointer to allocated memory. This can be newly allocated memory or re-use of previously
+		// returned object. Object is not initialized and shouldn't be considered to be in a valid state.
+		T* Allocate()
+		{
+			if (this->pool.empty())
+			{
+				return static_cast<T*>(std::malloc(sizeof(T)));
```

Comment:
It just allocates required memory, it is the responsibility of the caller to call constructors/destructors when using this object pool. I tried to make it such that it can be both generic and construct the object, kind of like std smart pointers do, but it turned out to be beyond my C++ level unfortunately.